node-notes-cli/app.js

```diff
@@ -0,0 +1,22 @@
+const fs = require('fs');
+const dataJson = require('./data.json');
+const { nextId } = require('./data.json');
+
+if (process.argv[2] === 'read') {
+  for (const property in dataJson.notes) {
+    console.log(`${property}: ${dataJson.notes[property]}`);
+  }
+} else if (process.argv[2] === 'create' && process.argv[3]) {
+  dataJson.notes[nextId] = process.argv[3];
+  dataJson.nextId++;
+} else if (process.argv[2] === 'update' && process.argv[3] < nextId && process.argv[4]) {
+  dataJson.notes[process.argv[3]] = process.argv[4];
```

Comment:
The `update` operation should only be able to modify existing notes, not create new ones. Keep in mind that, in addition to the user being able to supply invalid `id`s, the `delete` operation can remove notes.

Reply:
hi @TimelikeClosure, I updated it so "update" can only modify existing ones. Could you help clarify "user being able to supply invalid ids?"

Reply:
@brian-luong That means if the user does something like this, the note `5` should stay deleted:

```bash
node app.js delete 5
node app.js update 5 "back from the dead"
```

Reply:
Oh got it, thank you